diff --git a/hardware/libhardware_legacy/wifi/wifi.c b/hardware/libhardware_legacy/wifi/wifi.c
index 5c8a11f..7fa631d 100644
--- a/hardware/libhardware_legacy/wifi/wifi.c
+++ b/hardware/libhardware_legacy/wifi/wifi.c
@@ -70,7 +70,7 @@ static char primary_iface[PROPERTY_VALUE_MAX];
 // sockets is in
 
 #ifdef USES_TI_MAC80211
-#define P2P_INTERFACE			"p2p0"
+#define P2P_INTERFACE			"wlan0"
 struct nl_sock *nl_soc;
 struct nl_cache *nl_cache;
 struct genl_family *nl80211;
@@ -813,13 +813,6 @@ int wifi_start_supplicant(int p2p_supported)
         ALOGE("Wi-Fi entropy file was not created");
     }
 
-#ifdef USES_TI_MAC80211
-    if (p2p_supported && add_remove_p2p_interface(1) < 0) {
-        ALOGE("Wi-Fi - could not create p2p interface");
-        return -1;
-    }
-#endif
-
     /* Clear out any stale socket files that might be left over. */
     wpa_ctrl_cleanup();
 
@@ -888,13 +881,6 @@ int wifi_stop_supplicant(int p2p_supported)
         return 0;
     }
 
-#ifdef USES_TI_MAC80211
-    if (p2p_supported && add_remove_p2p_interface(0) < 0) {
-        ALOGE("Wi-Fi - could not remove p2p interface");
-        return -1;
-    }
-#endif
-
     property_set("ctl.stop", supplicant_name);
     sched_yield();
 
